src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,138 @@
+     +     * @return the instantiated {@link SphereClient}.
+     +     */
+    protected static SphereClient createClient(@Nonnull final SphereClientConfigOptions clientConfigOptions) {
+        final SphereClient underlyingClient = createSphereClient(clientConfigOptions.getClientConfig());
+        return decorateSphereClient(underlyingClient, clientConfigOptions.getMaxRetries(),
+                context -> calculateDurationWithExponentialRandomBackoff(context.getAttempt(),
+                        clientConfigOptions.getInitialDelay(), clientConfigOptions.getTimeout()), clientConfigOptions.getMaxParallelRequests());
+    }
+
+    /**
+     * Creates a {@link BlockingSphereClient} with a custom {@code timeout} with a custom {@link TimeUnit}.
+     *
+     * @param clientConfigOptions the client configuration for the client with custom Timeout and TimeUnit.
+     * @return the instantiated {@link BlockingSphereClient}.
+     */
+    protected static SphereClient createBlockingSphereClient(@Nonnull final SphereClientConfigOptions clientConfigOptions) {
+
+        final SphereClient underlyingClient = createSphereClient(clientConfigOptions.getClientConfig());
+        final SphereClient decoratedClient = decorateSphereClient(underlyingClient, clientConfigOptions.getMaxRetries(),
+                context -> calculateDurationWithExponentialRandomBackoff(context.getAttempt(),
+                        clientConfigOptions.getInitialDelay(), clientConfigOptions.getTimeout()), clientConfigOptions.getMaxParallelRequests());
+
+        return BlockingSphereClient.of(decoratedClient, clientConfigOptions.getTimeout(), clientConfigOptions.getTimeUnit());
+    }
+
+
+    protected static SphereClient createSphereClient(@Nonnull final SphereClientConfig clientConfig) {
+        final HttpClient httpClient = getHttpClient();
+        final SphereAccessTokenSupplier tokenSupplier =
+                SphereAccessTokenSupplier.ofAutoRefresh(clientConfig, httpClient, false);
+        return SphereClient.of(clientConfig, httpClient, tokenSupplier);
+    }
+
+    protected static SphereClient decorateSphereClient(
+            @Nonnull final SphereClient underlyingClient,
+            @Nonnull final long maxRetryAttempt,
+            @Nonnull final Function<RetryContext, Duration> durationFunction,
+            @Nonnull final int maxParallelRequests) {
+
+        final SphereClient retryClient = withRetry(underlyingClient, maxRetryAttempt, durationFunction);
+        return withLimitedParallelRequests(retryClient, maxParallelRequests);
+    }
+
+        /**
+         * Gets an asynchronous {@link HttpClient} to be used by the {@link BlockingSphereClient}.
+         *
+         * @return {@link HttpClient}
+         */
+    private static HttpClient getHttpClient() {
+        final AsyncHttpClient asyncHttpClient =
+                new DefaultAsyncHttpClient(
+                        new DefaultAsyncHttpClientConfig.Builder().build());
```

Comment:
the timeouts in here could be configurable or we could also expose an option to change the underlying client so it could be configured, check my comment: https://github.com/commercetools/commercetools-sync-java/pull/609#discussion_r516596428